### DISCUSSION TITLE
More customizable

### DISCUSSION THREAD

**Discussion by njfamirm** [telegraf/telegraf#1796] [Q&A]
Hi,
First of all, thank you for this great library.
Especially because of the excellent custom context feature.

I was able to use this feature to handle many things that made my code very simple.

Now I need two functions to improve my code,
One is customTelegram class
https://github.com/telegraf/telegraf/blob/a21fa139897fc27744fd0da1bab7a01d23d590b2/src/telegram.ts#L9

Another thing is how to pass parameters to customContext constructors?

**Comment by MKRhere** [comment]
This is not possible. Context is instantiated by Telegraf, and you cannot control it. Maybe you can do builder pattern here.

> Now I need two functions to improve my code

I actually don't think you _need_ it, so much as you think you need it.
